backend/prisma/schema.prisma

```diff
@@ -105,3 +156,40 @@ model SolutionAnalysis {
 
   solution Solution @relation(fields: [solutionId], references: [id], onDelete: Cascade)
 }
+
+// Tables to keep intermediate student activity:
```

Comment:
This is put in place to collect data for WP3.
I'd appreciate some early feedback.